### PR TITLE
prost: Pre-compile the tool

### PR DESCRIPTION
**Pull request by NickeZ**
Compiling the tool at build time increases build time and complicates
things when we want to support "cargo test" in the rust libraries.

### CODE REVIEW

File: messages/CMakeLists.txt
Comment (benma) [inline messages/CMakeLists.txt:59]: Repo moved, the comment is now here: https://github.com/tokio-rs/prost/issues/344#issuecomment-650721245

It seems the relevant [bug](https://github.com/rust-lang/cargo/issues/7915) has been fixed and [stabilized in 1.51](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#features).

I think this means we can delete the prost-build altogether and use prost as it is documented, without using this custom tool as a workaround.

Comment (NickeZ) [reply messages/CMakeLists.txt:59]: Yeah, I made a new PR.

**Comment by NickeZ** [comment]
Reopened. As I said in #831: #819 depends on either #829 or #831.

**Comment by benma** [comment]
Let's go with this one, seems to have many benefits over the other (faster build times, no explosion in the vendor folder, etc).

I build and pushed the Dockerfile with the changes of this PR as tag 22. If you rebase and point CI to `:22` like here, CI here should pass:

https://github.com/digitalbitbox/bitbox02-firmware/pull/839/files

(or wait until that PR is merged and then rebase).

**Review by benma — APPROVED** [review]
tACK, but best to wait until https://github.com/digitalbitbox/bitbox02-firmware/pull/839/files is merged and then rebase, so the CI here can pass.

**Comment by NickeZ** [comment]
rebased

**Pull request merged**
